src/IO/ConnectionTimeouts.h

```diff
@@ -72,6 +58,57 @@ struct ConnectionTimeouts
     static ConnectionTimeouts getFetchPartHTTPTimeouts(const ServerSettings & server_settings, const Settings & user_settings);
 
     ConnectionTimeouts getAdaptiveTimeouts(const String & method, bool first_attempt, bool first_byte) const;
+
+#define DECLARE_BUILDER_FOR_MEMBER(member, setter_func) \
+    ConnectionTimeouts & setter_func(size_t seconds); \
+    ConnectionTimeouts & setter_func(Poco::Timespan span); \
+
+APPLY_FOR_ALL_CONNECTION_TIMEOUT_MEMBERS(DECLARE_BUILDER_FOR_MEMBER)
+#undef DECLARE_BUILDER_FOR_MEMBER
+
+    ConnectionTimeouts & withAnyConnectionTimeout(size_t seconds);
+    ConnectionTimeouts & withAnyConnectionTimeout(Poco::Timespan span);
 };
 
+#define DEFINE_BUILDER_FOR_MEMBER(member, setter_func) \
+    inline ConnectionTimeouts & ConnectionTimeouts::setter_func(size_t seconds) \
+    { \
+        return setter_func(Poco::Timespan(seconds, 0)); \
+    } \
+    inline ConnectionTimeouts & ConnectionTimeouts::setter_func(Poco::Timespan span) \
+    { \
+        member = span \
+        return *this; \
+    } \
+                                                            \
+    APPLY_FOR_ALL_CONNECTION_TIMEOUT_MEMBERS(DEFINE_BUILDER_FOR_MEMBER)
+
+#undef DEFINE_BUILDER_FOR_MEMBER
+
+
+inline ConnectionTimeouts ConnectionTimeouts::getSaturated(Poco::Timespan limit) const
+{
+#define SATURATE_MEMBER(member, setter_func) \
+    .setter_func(saturate(member, limit))
+
+    return ConnectionTimeouts(*this)
+APPLY_FOR_ALL_CONNECTION_TIMEOUT_MEMBERS(SATURATE_MEMBER);
+
+#undef SATURETE_MEMBER
+}
+
+#undef APPLY_FOR_ALL_CONNECTION_TIMEOUT_MEMBERS
+
+inline ConnectionTimeouts & ConnectionTimeouts::withAnyConnectionTimeout(size_t seconds)
```

Comment:
I'm having some concerns that people will automatically use `withConnectionTimeout` by default and not set `secure_connection_timeout` by accident
the only time you want to just set `connection_timeout` explicitly is if you want to set the `secure_connection_timeout` so maybe have `withConnectionTimeout` set both timeouts? and you can just override secure timeout with `withSslConnectionTimeout`.